Imagekit/Imagekit.cs

```diff
@@ -1,473 +1,341 @@
-        {
-            Transformation transformation = (Transformation)options["transformation"];
-            string tranformationString = transformation.Generate();
-            return new Url(options).UrlBuilder(tranformationString);
-        }
-
-
-        public Dictionary<string, string> getUploadData(AuthParamResponse clientAuth = null)
-        {
-            if (!options.ContainsKey("fileName") || string.IsNullOrEmpty((string)options["fileName"]))
-            {
-                throw new ArgumentException(errorMessages.MISSING_UPLOAD_FILENAME_PARAMETER);
-            }
-
-            Dictionary<string, string> postData = new Dictionary<string, string>();
-            postData.Add("fileName", (string)options["fileName"]);
-            if (options.ContainsKey("folder") && !string.IsNullOrEmpty((string)options["folder"]))
-            {
-                postData.Add("folder", (string)options["folder"]);
-            }
-            if (options.ContainsKey("isPrivateFile") && (bool)options["isPrivateFile"] == true)
-            {
-                postData.Add("isPrivateFile", "true");
-            }
-            if (options.ContainsKey("useUniqueFileName") && (bool)options["useUniqueFileName"] == false)
-            {
-                postData.Add("useUniqueFileName", "false");
-            }
-            if (options.ContainsKey("customCoordinates") && !string.IsNullOrEmpty((string)options["customCoordinates"]))
-            {
-                postData.Add("customCoordinates", (string)options["customCoordinates"]);
-            }
-            if (options.ContainsKey("responseFields") && !string.IsNullOrEmpty((string)options["responseFields"]))
-            {
-                postData.Add("responseFields", (string)options["responseFields"]);
-            }
-            if (options.ContainsKey("tags") && !string.IsNullOrEmpty((string)options["tags"]))
-            {
-                postData.Add("tags", (string)options["tags"]);
-            }
-            else if (options.ContainsKey("tagsList"))
-            {
-                var tags = (string[])options["tagsList"];
-                if (tags.Any())
-                {
-                    postData.Add("tags", string.Join(",", tags));
-                }
-            }
-            if (clientAuth != null)
-            {
-                postData.Add("signature", clientAuth.signature);
-                postData.Add("expire", clientAuth.expire.ToString());
-                postData.Add("token", clientAuth.token);
-                postData.Add("publicKey", (string)options["publicKey"]);
-            }
-            return postData;
+        private RestClient restClient;
+
+        public RestClient RestClient { get => this.restClient; set => this.restClient = value; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ImageKit"/> class.
+        /// </summary>
+        public ImageKitClient(string publicKey ,string privateKey, string apiBaseUrl)
```

Comment:
The third option in imageKit client must be "URL endpoint" of an account, not the api URL. URL endpoint is like .. "https://ik.imagekit.io/imagekitId/". 

API url can be defined in the code. https://api.imagekit.io for all media apis and https://upload.imagekit.io for all upload api requests.